hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/TargetManagement.java

```diff
@@ -248,13 +247,13 @@ Long countTargetByFilters(Collection<TargetUpdateStatus> status, Boolean overdue
      *            the pageRequest to enhance the query for paging and sorting
      * @param distributionSetId
      *            id of the {@link DistributionSet}
-     * @param targetFilterQuery
-     *            {@link TargetFilterQuery}
+     * @param rsqlParam
+     *            filter definition in RSQL syntax
      * @return a page of the found {@link Target}s
      */
     @PreAuthorize(SpringEvalExpressions.HAS_AUTH_READ_TARGET)
     Page<Target> findAllTargetsByTargetFilterQueryAndNonDS(@NotNull Pageable pageRequest, Long distributionSetId,
-            @NotNull TargetFilterQuery targetFilterQuery);
+            @NotNull String rsqlParam);
```

Comment:
Good question. We use normally NotNull for these but I guess we might switch to not empty one. However, I am not fully sure concerning the consequences.